core/src/main/java/org/apache/calcite/sql/dialect/PhoenixSqlDialect.java

```diff
@@ -17,22 +17,51 @@
 package org.apache.calcite.sql.dialect;
 
 import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeSystem;
+import org.apache.calcite.rel.type.RelDataTypeSystemImpl;
 import org.apache.calcite.sql.SqlAlienSystemTypeNameSpec;
 import org.apache.calcite.sql.SqlDataTypeSpec;
 import org.apache.calcite.sql.SqlDialect;
 import org.apache.calcite.sql.SqlNode;
 import org.apache.calcite.sql.parser.SqlParserPos;
 import org.apache.calcite.sql.type.AbstractSqlType;
+import org.apache.calcite.sql.type.SqlTypeName;
 
 import org.checkerframework.checker.nullness.qual.Nullable;
 
 /**
  * A <code>SqlDialect</code> implementation for the Apache Phoenix database.
  */
 public class PhoenixSqlDialect extends SqlDialect {
+  public static final RelDataTypeSystem TYPE_SYSTEM =
```

Comment:
are there links for these other type systems too?

Reply:
Done, had added links for other systems, thanks @mihaibudiu